fxa/crypto.py

```diff
@@ -128,3 +132,25 @@ def unbundle(key, namespace, payload):
     # XOR-decrypt the ciphertext using the derived key.
     xor_key = key_material[32:]
     return xor(xor_key, ciphertext)
+
+
+def generate_keypair():
+    """Generate a new DSA keypair for use with PyBrowserID.
+
+    This function returns a tuple (public_data, private_key) giving the
+    JSON-serializable public-key data and the associated private key as a
+    browserid.jwt.Key object.
+    """
+    key = dsa.generate_private_key(1024, backend=default_backend())
+    params = key.parameters().parameter_numbers()
+    data = {
+        "algorithm": "DS",
+        "p": to_hex(params.p),
+        "q": to_hex(params.q),
+        "g": to_hex(params.g),
+        "y": to_hex(key.public_key().public_numbers().y),
+        "x": to_hex(key.private_numbers().x),
+    }
+    private_key = browserid.jwt.DS128Key(data)
+    del data["x"]
+    return (data, private_key)
```

Comment:
I must be an oddball - I really like explicit parens return returning a tuple, as an easy and obvious visual reminder that's you're returning more than one value.  But consistency FTW I guess, I'll get rid of them...